litgpt/chat/base.py

```diff
@@ -196,6 +197,10 @@ def main(
         compile: Whether to use compilation to speed up token generation. Will increase startup time.
         multiline: Whether to support multiline input prompts.
     """
+    if not checkpoint_dir.is_dir():
```

Comment:
For me it looks like this if statement and the code don't belong to each other.
I guess it's better to add this prefix if the path doesn't start with "checkpoints".
In other words a path that starts with "checkpoints" doesn't guarantee that the directory exists.

Reply:
Thanks for the review!

> I guess it's better to add this prefix if the path doesn't start with "checkpoints".
In other words a path that starts with "checkpoints" doesn't guarantee that the directory exists.

That's true. If the `"checkpoints" / checkpoint_dir` doesn't exist or contains invalid files, it will still check that in the 

```python
check_valid_checkpoint_dir(checkpoint_dir)
```

below.

The use case we want to cover is the following:

```bash
# ligpt [action] [model]
litgpt  download  meta-llama/Meta-Llama-3-8B-Instruct # downloads files to checkpoints/...
litgpt  chat      meta-llama/Meta-Llama-3-8B-Instruct
litgpt  finetune  meta-llama/Meta-Llama-3-8B-Instruct
litgpt  pretrain  meta-llama/Meta-Llama-3-8B-Instruct
litgpt  serve     meta-llama/Meta-Llama-3-8B-Instruct
```

> For me it looks like this if statement and the code don't belong to each other.

Could you explain a bit more what you mean here?

Reply:
> Could you explain a bit more what you mean here?

You check whether the folder exists. If it doesn't, then you prepend "checkpoints" and it looks like it should fix the non existing issue. But it's not guaranteed.

So, from my pov, it's better to check whether the path starts with "checkpoints" and if it doesn't - prepend it. Looks more cohesive to me.

Reply:
I agree and added this check in the mean time in the previous commits